pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -744,6 +792,117 @@ func applyLoadBalancer(cluster *apiv2.Cluster, lb *networking.LoadBalancerSettin
 			},
 		}
 	}
+
+	// Locality weighted load balancing
+	cluster.CommonLbConfig = &apiv2.Cluster_CommonLbConfig{
+		LocalityConfigSpecifier: &apiv2.Cluster_CommonLbConfig_LocalityWeightedLbConfig_{
+			LocalityWeightedLbConfig: &apiv2.Cluster_CommonLbConfig_LocalityWeightedLbConfig{},
+		},
+	}
+}
+
+func applyLocalityLBSetting(
+	proxy *model.Proxy,
+	loadAssignment *apiv2.ClusterLoadAssignment,
+	localityLB *meshconfig.LocalityLoadBalancerSetting,
+	localityPriority bool) {
+	if proxy == nil || loadAssignment == nil {
+		return
+	}
+
+	// one of Distribute or Failover settings can be applied.
+	if localityLB.GetDistribute() != nil {
+		applyLocalityWeight(proxy, loadAssignment, localityLB.GetDistribute())
+	} else if localityPriority {
+		applyLocalityFailover(proxy, loadAssignment, localityLB.GetFailover())
+	}
+}
+
+func applyLocalityWeight(
+	proxy *model.Proxy,
+	loadAssignment *apiv2.ClusterLoadAssignment,
+	distribute []*meshconfig.LocalityLoadBalancerSetting_Distribute) {
+	if distribute == nil {
+		return
+	}
+
+	for _, localityWeightSetting := range distribute {
```

Comment:
it would be good if you added some text on what envoy expects and what is the general algorithm below

Reply:
ok